packages/search-field/src/SearchField.js

```diff
@@ -37,9 +37,17 @@ class SearchField extends React.Component {
     this.props.onSearchChange,
     this.props.debounceDelay
   );
+
   onSearchChange = event => {
     const { value } = event.target;
 
+    // We are forcing an update to render an input,
+    // otherwise we'd have a div instead,
+    // and calling focus on it
+    if (value === "") {
+      this.forceUpdate(() => this.input.focus());
```

Comment:
This is difficult to understand, even with the comment. The focus is necessary after a click on the pencil, that reason should be mentioned, or alternatively the focus should be an option to pass to the input element upon mounting, I have pasted an example on slack.

Reply:
Talking to @vNakamura we liked your suggestion as it's clearer about what's happening. Please see the next commit.

We also found and fixed another related bug, which was happening when typing something and then clicking anywhere outside the typing area: the component blinks (caused by input hoc) and the text is cleared. See gif.

![search-focus-v3](https://user-images.githubusercontent.com/17071199/41988251-e6be8208-7a11-11e8-89aa-29f82457a119.gif)